### PR TITLE
(Partially) add type hints. See #309. 

### PR DESCRIPTION
**Pull request by AdamHillier**

### CODE REVIEW

File: larq/constraints.py [hunk 1]
```diff
@@ -34,13 +36,13 @@ class WeightClip(tf.keras.constraints.Constraint):
...
-    def __init__(self, clip_value=1):
+    def __init__(self, clip_value: float = 1):
         self.clip_value = clip_value
 
-    def __call__(self, x):
+    def __call__(self, x: tf.Tensor) -> tf.Tensor:
         return tf.clip_by_value(x, -self.clip_value, self.clip_value)
 
-    def get_config(self):
+    def get_config(self) -> Dict:
```
Comment (lgeiger) [inline larq/constraints.py]: General Python typehints question: Is using the builtin `dict` equivalent to `typings.Dict` in this case?
```suggestion
    def get_config(self) -> dict:
```

File: larq/models.py [hunk 2]
```diff
@@ -415,9 +435,6 @@ def summary(model, print_fn=None, include_macs=True):
             "`input_shape` argument in the first layer(s) for automatic build."
         )
 
-    if print_fn is None:
-        print_fn = print
-
```
Comment (lgeiger) [inline larq/models.py]: I think we still need this.

Comment (AdamHillier) [reply larq/models.py]: I made it the default value of the arg above so I don't think so?

https://github.com/larq/larq/blob/246a64a4d0ce7f3044cafee58dbb04b42b94879b/larq/models.py#L395

Comment (lgeiger) [reply larq/models.py]: Ahh sorry, missed that. I think there was a reason, why I didn't do that in the first place. But I can't remember and CI is happy with it :+1:

File: larq/constraints.py (hunk 1 above)
Comment (AdamHillier) [reply larq/constraints.py]: Not 100% sure about the difference, but now I look into it apparently it's better to use `typing.Mapping` for such things, so I will edit.

**Pull request merged**